File: src/org/ggp/base/util/ui/ChainingReplacedElementFactory.java
package org.ggp.base.util.ui;

/* http://www.samuelrossille.com/home/render-html-with-svg-to-pdf-with-flying-saucer.html */

import java.util.ArrayList;
import java.util.List;

import org.w3c.dom.Element;
import org.xhtmlrenderer.extend.ReplacedElement;
import org.xhtmlrenderer.extend.ReplacedElementFactory;
import org.xhtmlrenderer.extend.UserAgentCallback;
import org.xhtmlrenderer.layout.LayoutContext;
import org.xhtmlrenderer.render.BlockBox;
import org.xhtmlrenderer.simple.extend.FormSubmissionListener;

class ChainingReplacedElementFactory(ReplacedElementFactory):
    private List<ReplacedElementFactory> replacedElementFactories = new ArrayList<ReplacedElementFactory>();

    def void addReplacedElementFactory(ReplacedElementFactory replacedElementFactory):
    replacedElementFactories.add(0, replacedElementFactory);
    }


    def ReplacedElement createReplacedElement(LayoutContext c, BlockBox box, UserAgentCallback uac, int cssWidth, int cssHeight):
    for(ReplacedElementFactory replacedElementFactory : replacedElementFactories):
	    ReplacedElement element = replacedElementFactory.createReplacedElement(c, box, uac, cssWidth, cssHeight);
	    if(element != null):
        return element;
	    }
    return null;
    }

    def void reset():
    for(ReplacedElementFactory replacedElementFactory : replacedElementFactories):
	    replacedElementFactory.reset();
    }

    def void remove(Element e):
    for(ReplacedElementFactory replacedElementFactory : replacedElementFactories):
	    replacedElementFactory.remove(e);
    }

    def void setFormSubmissionListener(FormSubmissionListener listener):
    for(ReplacedElementFactory replacedElementFactory : replacedElementFactories):
	    replacedElementFactory.setFormSubmissionListener(listener);
    }

File: ggpy/cruft/autocode/ChainingReplacedElementFactory.py
#!/usr/bin/env python
""" generated source for module ChainingReplacedElementFactory """
# package: org.ggp.base.util.ui
#  http://www.samuelrossille.com/home/render-html-with-svg-to-pdf-with-flying-saucer.html 
import java.util.ArrayList

import java.util.List

import org.w3c.dom.Element

import org.xhtmlrenderer.extend.ReplacedElement

import org.xhtmlrenderer.extend.ReplacedElementFactory

import org.xhtmlrenderer.extend.UserAgentCallback

import org.xhtmlrenderer.layout.LayoutContext

import org.xhtmlrenderer.render.BlockBox

import org.xhtmlrenderer.simple.extend.FormSubmissionListener

class ChainingReplacedElementFactory(ReplacedElementFactory):
    """ generated source for class ChainingReplacedElementFactory """
    replacedElementFactories = ArrayList()

    def addReplacedElementFactory(self, replacedElementFactory):
        """ generated source for method addReplacedElementFactory """
        self.replacedElementFactories.add(0, replacedElementFactory)

    def createReplacedElement(self, c, box, uac, cssWidth, cssHeight):
        """ generated source for method createReplacedElement """
        for replacedElementFactory in replacedElementFactories:
            if element != None:
                return element
        return None

    def reset(self):
        """ generated source for method reset """
        for replacedElementFactory in replacedElementFactories:
            replacedElementFactory.reset()

    def remove(self, e):
        """ generated source for method remove """
        for replacedElementFactory in replacedElementFactories:
            replacedElementFactory.remove(e)

    def setFormSubmissionListener(self, listener):
        """ generated source for method setFormSubmissionListener """
        for replacedElementFactory in replacedElementFactories:
            replacedElementFactory.setFormSubmissionListener(listener)

